### ISSUE TITLE
ScalaTest for Scala 3 depends on the Scala 2 version of scala-xml

### ISSUE THREAD

**Issue by SethTisue** [scalatest/scalatest#1973]
This can cause problems for downstream users, along the lines of this, recently reported on sbt/sbt gitter (cc @DarkFrog26):

```
[error] Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:/home/mhicks/projects/open/fabric/"), "parseJVM"):
[error]    org.scala-lang.modules:scala-xml _3.0.0-RC1, _2.13
```

scala-xml is available for Scala 3, but admittedly only in milestones, as per https://github.com/scala/scala-xml/issues/432

regardless, I would suggest you go ahead and depend on the milestones, on the assumption that by the time Scala 3.0 final comes out, there will be also be a final scala-xml 2.0 release (see linked ticket)

**Comment by bvenners** [comment]
@SethTisue OK. I've been wanting to get rid of the dependency on Scala XML, but hadn't done so yet. If that's not too involved maybe we'll solve it that way. Either way we'll make a new release.

**Comment by thesamet** [comment]
This became an error for ScalaPB when updating scalacheck-1.15 to 3.2.6.0 (https://github.com/scalapb/ScalaPB/pull/1084/checks?check_run_id=2054571558)

**Comment by SethTisue** [comment]
@thesamet probably you have some _other_ dependency that has a transitive dependency on the 2.13 version of scala-xml

**Comment by thesamet** [comment]
Thanks @SethTisue . Indeed we have a `withDottyCompat` dependency on coursier which depends on scalaxml. I wonder why this error is only triggered when updating scalatest to 3.2.6.0. The error isn't triggered with 3.2.5.0.

**Comment by thesamet** [comment]
Minimal example: https://github.com/thesamet/scalaxml-issue, downgrade to 3.2.5.0 to make the error go away.

**Comment by bvenners** [comment]
I think it happens in 3.2.6 we changed the dependency to use scala-xml 2.0.0-M5 for Scala 3 build and 1.3.0 for Scala 2 build. In 3.2.5 ScalaTet is probably using the same scala-xml as coursier.

**Comment by SethTisue** [comment]
@thesamet you'll need to exclude the transitive dependency from coursier when you `withDottyCompat` it

**Comment by SethTisue** [comment]
is this fixed as of 3.2.7?

**Comment by bvenners** [comment]
@seth yes i believe so. I will double check later today and confirm.

**Comment by cheeseng** [comment]
@SethTisue @bvenners Yes it is released with 3.2.6, so 3.2.7 should have the fix too.
